File: Ejercicio3.java
package ape02;
import java.util.Scanner;

public class Jean {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int codigo;
        String color;
        String talla;
        char tenido;
        int cantidadTenidos;
        double precio;
        int cantidadBotones;
        double humedad;
        String estadoTela;
        int opcion;

        System.out.println("REGISTRO DEL JEAN");

        System.out.print("Ingrese el codigo: ");
        codigo = sc.nextInt();

        sc.nextLine();

        System.out.print("Ingrese el color: ");
        color = sc.nextLine();

        System.out.print("Ingrese la talla: ");
        talla = sc.nextLine();

        System.out.print("Fue tenido? (S/N): ");
        tenido = sc.next().charAt(0);

        System.out.print("Ingrese la cantidad de tenidos: ");
        cantidadTenidos = sc.nextInt();

        System.out.print("Ingrese el precio: $");
        precio = sc.nextDouble();

        System.out.print("Ingrese la cantidad de botones: ");
        cantidadBotones = sc.nextInt();

        System.out.print("Ingrese la humedad de la tela: ");
        humedad = sc.nextDouble();

        sc.nextLine();

        System.out.print("Ingrese el estado de la tela: ");
        estadoTela = sc.nextLine();

        do {

            System.out.println("\nMENU");
            System.out.println("1. Mostrar datos");
            System.out.println("2. Lavar");
            System.out.println("3. Secar");
            System.out.println("4. Salir");
            System.out.print("Seleccione una opcion: ");
            opcion = sc.nextInt();

            switch (opcion) {

                case 1:

                    System.out.println("\nDATOS DEL JEAN");
                    System.out.println("Codigo: " + codigo);
                    System.out.println("Color: " + color);
                    System.out.println("Talla: " + talla);
                    System.out.println("Fue tenido: " + tenido);
                    System.out.println("Cantidad de tenidos: "
                            + cantidadTenidos);
                    System.out.printf("Precio: $%.2f%n", precio);
                    System.out.println("Cantidad de botones: "
                            + cantidadBotones);
                    System.out.println("Humedad: " + humedad);
                    System.out.println("Estado de la tela: "
                            + estadoTela);

                    break;

                case 2:

                    if (cantidadTenidos > 0) {

                        cantidadTenidos--;

                        System.out.println("El jean fue lavado.");
                        System.out.println("Cantidad de tenidos: "
                                + cantidadTenidos);

                    } else {

                        System.out.println(
                                "No se puede disminuir mas la cantidad de tenidos.");
                    }

                    break;

                case 3:

                    if (humedad > 0) {

                        humedad -= 10;

                        if (humedad < 0) {
                            humedad = 0;
                        }

                        System.out.println("El jean fue secado.");
                        System.out.println("Humedad actual: "
                                + humedad);

                    } else {

                        System.out.println(
                                "El jean ya no tiene humedad.");
                    }

                    break;

                case 4:

                    System.out.println("Saliendo del sistema...");

                    break;

                default:

                    System.out.println("Opcion invalida.");
            }

        } while (opcion != 4);

        System.out.println("Programa finalizado.");

        sc.close();
    }
}

File: Ejercicio3.cpp
#include <iostream>
#include <iomanip>

using namespace std;

int main() {

    int codigo;
    string color;
    string talla;
    char tenido;
    int cantidadTenidos;
    double precio;
    int cantidadBotones;
    double humedad;
    string estadoTela;
    int opcion;

    cout << "REGISTRO DEL JEAN" << endl;

    cout << "Ingrese el codigo: ";
    cin >> codigo;

    cin.ignore();

    cout << "Ingrese el color: ";
    getline(cin, color);

    cout << "Ingrese la talla: ";
    getline(cin, talla);

    cout << "Fue tenido? (S/N): ";
    cin >> tenido;

    cout << "Ingrese la cantidad de tenidos: ";
    cin >> cantidadTenidos;

    cout << "Ingrese el precio: $";
    cin >> precio;

    cout << "Ingrese la cantidad de botones: ";
    cin >> cantidadBotones;

    cout << "Ingrese la humedad de la tela: ";
    cin >> humedad;

    cin.ignore();

    cout << "Ingrese el estado de la tela: ";
    getline(cin, estadoTela);

    do {

        cout << "\nMENU" << endl;
        cout << "1. Mostrar datos" << endl;
        cout << "2. Lavar" << endl;
        cout << "3. Secar" << endl;
        cout << "4. Salir" << endl;
        cout << "Seleccione una opcion: ";
        cin >> opcion;

        switch (opcion) {

            case 1:

                cout << "\nDATOS DEL JEAN" << endl;
                cout << "Codigo: " << codigo << endl;
                cout << "Color: " << color << endl;
                cout << "Talla: " << talla << endl;
                cout << "Fue tenido: " << tenido << endl;
                cout << "Cantidad de tenidos: "
                     << cantidadTenidos << endl;

                cout << fixed << setprecision(2);
                cout << "Precio: $" << precio << endl;

                cout << "Cantidad de botones: "
                     << cantidadBotones << endl;

                cout << "Humedad: " << humedad << endl;

                cout << "Estado de la tela: "
                     << estadoTela << endl;

                break;

            case 2:

                if (cantidadTenidos > 0) {

                    cantidadTenidos--;

                    cout << "El jean fue lavado." << endl;
                    cout << "Cantidad de tenidos: "
                         << cantidadTenidos << endl;

                } else {

                    cout << "No se puede disminuir mas "
                         << "la cantidad de tenidos."
                         << endl;
                }

                break;

            case 3:

                if (humedad > 0) {

                    humedad -= 10;

                    if (humedad < 0) {
                        humedad = 0;
                    }

                    cout << "El jean fue secado." << endl;
                    cout << "Humedad actual: "
                         << humedad << endl;

                } else {

                    cout << "El jean ya no tiene humedad."
                         << endl;
                }

                break;

            case 4:

                cout << "Saliendo del sistema..." << endl;

                break;

            default:

                cout << "Opcion invalida." << endl;
        }

    } while (opcion != 4);

    cout << "Programa finalizado." << endl;

    return 0;
}
